en/ros/external_position_estimation.md

```diff
@@ -1,114 +1,220 @@
+Reducing them will cause the estimator to trust the incoming pose estimate more. 
+You may need to set them lower than the allowed minimum and force-save. 
+
+> **Tip** If performance is still poor, try increasing the [LPE_PN_V](../advanced/parameter_reference.md#LPE_PN_V) parameter. 
+  This will cause the estimator to trust measurements more during velocity estimation.
+
+  
+## Working with ROS
+
+ROS is not *required* for supplying external pose information, but is highly recommended as it already comes with good integrations with VIO and MoCap systems.
+PX4 must already have been set up as above.
+
+### Getting pose data into ROS
+
+* Install the `vrpn_client_ros` package
+* You can get each rigid body pose on an individual topic by running
+  ```bash
+  roslaunch vrpn_client_ros sample.launch server:=<mocap machine ip>
+  ```
+
+If you named the rigidbody as `robot1`, you will get a topic like `/vrpn_client_node/robot1/pose`
+
+
+### Relaying Pose Data to PX4
+
+MAVROS has plugins to relay a visual estimation from a VIO or MoCap system using the following pipelines:
+
+ROS | MAVLink | uORB 
+--- | --- | ---
+/mavros/vision_pose/pose | [VISION_POSITION_ESTIMATE](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE) | `vehicle_visual_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_VISION_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_VISION_NED)) | `vehicle_visual_odometry`
+/mavros/mocap/pose | [ATT_POS_MOCAP](https://mavlink.io/en/messages/common.html#ATT_POS_MOCAP) | `vehicle_mocap_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_MOCAP_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_MOCAP_NED)) | `vehicle_mocap_odometry`
+
+You can use any of the above pipelines with LPE. 
 
-> **Tip**If performance is still poor, try increasing the `LPE_PN_V` parameter. This will cause the estimator to trust measurements more during velocity estimation..
+If you're working with EKF2, only the "vision" pipelines are supported. 
+To use MoCap data with EKF2 you will have to [remap](http://wiki.ros.org/roslaunch/XML/remap) the pose topic that you get from MoCap:
+- MoCap ROS topics of type `geometry_msgs/PoseStamped` or `geometry_msgs/PoseWithCovarianceStamped` must be remapped to `/mavros/vision_pose/pose`. 
+  The `geometry_msgs/PoseStamped` topic is most common as MoCap doesn't usually have associated covariances to the data.
+- If you get data through a `nav_msgs/Odometry` ROS message then you will need to remap it to `/mavros/odometry/odom`.
+- TBD from `/vrpn_client_node/<rigid_body_name>/pose` to `/mavros/vision_pose/pose`. 
```

Comment:
@TSC21 So this line was copied from optitrack stuff too. But it doesn't match your cases (in preceding bullets). Is this invalid, or just another case to cover?